weblate/static/loader-bootstrap.js

```diff
@@ -1525,8 +1525,15 @@ $(function () {
     $('.search-group li a').click(function () {
         var $this = $(this);
         var $button = $this.closest('.input-group-btn').find('button');
-        $button.data('field', $this.data('field'));
+        $button.attr('data-field', $this.data('field'));
         $button.find('span.search-label').text($this.text());
+        if ($this.closest('.query-field').length) {
+            if ($this.data('sort')) {
+                $('#id_sort_by').val($this.data('sort'));
+            } else {
+                $('#id_q').insertAtCaret(' ' + $this.data('field') + ' ');
```

Comment:
The approach of inserting the query is not really user-friendly - if you keep changing the options, the resulting query will not make much sense. I think it makes sense here to directly perform the query after clicking on the option.

Auto submit on sort also might be more friendly than this.

Reply:
>  I think it makes sense here to directly perform the query after clicking on the option.

I was thinking in the term like what if someone selects a query from dropdown but also wants to add their own search term to it? I mean in that case, they have to wait for the result and then enter the additional parts.

Reply:
> Auto submit on sort also might be more friendly than this.

This I kind of do agree with. I will update this.

Reply:
> I was thinking in the term like what if someone selects a query from dropdown but also wants to add their own search term to it? I mean in that case, they have to wait for the result and then enter the additional parts.

I consider this as a selection of predefined queries. To build custom query use the input box. This is how it also behaves on Sentry and I quite like it there as an user (the left dropdown was not shown in https://github.com/WeblateOrg/weblate/issues/3308#issuecomment-617066002).

The problem with adding strings is that you quickly end up with query that doesn't make sense.

Also, the auto submission avoids weird effect of changing width of the text input selecting the values.

Reply:
Sounds good. Just one question. Let's say someone made a query using the query field before, pressed search, got the result. Now when I click on the dropdown list (or sorting), the query term is still taken into consideration when fetching the new results, right?

Reply:
In Sentry it completely discards the current query and it was never an issue for me.